### PR TITLE
Return `undefined` `loc` if declaration missing `source`

### LINKED ISSUE #6898: Parcel 2 - issue with tailwindcss and JIT only when using @tailwindcss/forms
As soon as I include @tailwindcss/forms into my tailwind config, I am faced with a compilation error in dev and build mode.

This does not work (compilation fail)

```
module.exports = {
  mode: 'jit',
  purge: ["./src/*.{html,js}"],
  darkMode: 'media',
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [require('@tailwindcss/forms')],
};
```

If you don't include the tailwind forms no issues.

This is fine, no issues:

```
module.exports = {
  mode: 'jit',
  purge: ["./src/*.{html,js}"],
  darkMode: 'media',
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
};
```

If you don't enable JIT no issues.

This also is fine, no issues:

```
module.exports = {
  purge: ["./src/*.{html,js}"],
  darkMode: 'media',
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [require('@tailwindcss/forms')],
};
```

# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

https://github.com/sync/tailwind-jit-forms

## 🤔 Expected Behavior

running parcel or parcel build should not throw an error.

## 😯 Current Behavior

Parcel fails to complete the build.

```
🚨 Build failed.

@parcel/transformer-css: Cannot read property 'start' of undefined

  TypeError: Cannot read property 'start' of undefined
  at /Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/@parcel/transformer-css/lib/CSSTransformer.js:213:71
  at walk (/Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss-value-parser/lib/walk.js:7:16)
  at ValueParser.walk (/Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss-value-parser/lib/index.js:18:3)
  at /Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/@parcel/transformer-css/lib/CSSTransformer.js:209:16
  at /Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:72:18
  at /Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:55:18
  at Rule.each (/Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:41:16)
  at Rule.walk (/Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:52:17)
  at /Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:60:24
  at Root.each (/Users/anthonymittaz/Projects/Parcel/tailwind-jit-forms/node_modules/postcss/lib/container.js:41:16)
```

## 💁 Possible Solution

Tailwind JIT works fine, just facing this issue when adding the `@tailwindcss/forms` plugin into the mix

## 💻 Code Sample

https://github.com/sync/tailwind-jit-forms

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-rc.0
| Node             | Tested with Node 14 and 16 latest (v16.5.0)
| npm/Yarn         | latest yarn v1 1.22.10
| Operating System | macOS Monterey

### PR DESCRIPTION
**Pull request by sibbng**
# ↪️ Pull Request

Fixes #6898

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by DeMoorJasper** [comment]
Tests seem to be failing, could you fix them?

**Comment by sibbng** [comment]
> Tests seem to be failing, could you fix them?

Fixed. Remaining failures related to yarn pnp seems persist in `v2` branch too.

File: packages/transformers/css/src/CSSTransformer.js [hunk 1]
```diff
@@ -143,12 +143,15 @@ export default (new Transformer({
...
-              ),
+              loc:
+                decl.source && decl.source.start
+                  ? createLoc(
+                      decl.source.start,
+                      node.nodes[0].value,
+                      0,
+                      node.nodes[0].sourceIndex,
+                    )
+                  : createLoc({line: 1, column: 1}, ''),
```
Comment (mischnic) [inline packages/transformers/css/src/CSSTransformer.js]: I think `loc` should rather be `undefined` if we can't determine the position.

**Comment by mischnic** [comment]
Last thing: can you remove the `packages/core/integration-tests/test/integration/postcss-plugins-decl/index.js` and use `index.js` directly as the entry point (and remove `await run(b)`)?

**Comment by sibbng** [comment]
> and use `index.js` directly as the entry point

I guess, you meant to say `index.css` here.

**Comment by mischnic** [comment]
Yes 👍

**Review by mischnic — APPROVED** [review]

**Pull request merged**